fix: upgrade cudarc to 0.17.1

da-x opened the pull request:
Fixes the build with `--features cuda,dynamo-llm/block-manager`.

cudarc failed over: "Both dynamic-loading and dynamic-linking features are active, this is a bug"

Explanation:

- Before dbe48a1d2b9, mistral linked to cudarc 0.13.9 with dynamic _linking_, while Dynamo linked to 0.16.2 with dynamic _loading_. So they didn't see each other in the build process and it somehow worked.

- With the consolidation to 0.16.6 a single cudarc is used, and therefore feature selection intersects with this conflict.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Chores**
  * Updated internal dependencies to include additional features for improved compatibility and performance. No visible changes to end-users.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ryanolson commented:
Thanks!

rmccorm4 reviewed (commented):
Hi @da-x , thanks for the contribution. Can you take a look a the failed check?

ex:
```
#41 373.2   = note: /opt/rh/gcc-toolset-14/root/usr/libexec/gcc/x86_64-redhat-linux/14/ld: cannot find -lcuda: No such file or directory
#41 373.2           /opt/rh/gcc-toolset-14/root/usr/libexec/gcc/x86_64-redhat-linux/14/ld: cannot find -lnvrtc: No such file or directory
#41 373.2           /opt/rh/gcc-toolset-14/root/usr/libexec/gcc/x86_64-redhat-linux/14/ld: cannot find -lcurand: No such file or directory
#41 373.2           /opt/rh/gcc-toolset-14/root/usr/libexec/gcc/x86_64-redhat-linux/14/ld: cannot find -lcublas: No such file or directory
#41 373.2           /opt/rh/gcc-toolset-14/root/usr/libexec/gcc/x86_64-redhat-linux/14/ld: cannot find -lcublasLt: No such file or directory
#41 373.2           /opt/rh/gcc-toolset-14/root/usr/libexec/gcc/x86_64-redhat-linux/14/ld: cannot find -lcudart: No such file or directory
#41 373.2           collect2: error: ld returned 1 exit status
#41 373.2           
#41 373.2 
#41 373.2 error: could not compile `router` (bin "router") due to 1 previous error
#41 ERROR: process "/bin/sh -c cargo build \t--release \t--locked \t--features dynamo-llm/block-manager \t--workspace" did not complete successfully: exit code: 101
```

da-x commented:
The error was due to missing CUDA libraries in the Python wheel builder.

Ended up doing this differently. We can regain this back by upgrading to a different cudarc major, as other methods require less trivial changes in docker images with regard to CUDA library installation (having both cudarc uses with
`dynamic-linking` approach).

rmccorm4 commented:
Looking into recent changes to merge requirements for external contributors

<img width="284" height="67" alt="image" src="https://github.com/user-attachments/assets/994235de-2520-44d9-883b-b674d25bc991" />

da-x committed:
fix: upgrade cudarc to 0.17.1

Fix build with `--features cuda,dynamo-llm/block-manager`.

cudarc failed over: "Both dynamic-loading and dynamic-linking features
are active, this is a bug"

Explanation:

- Before dbe48a1d2b9, mistral linked to cudarc 0.13.9 with dynamic
  _linking_, while Dynamo linked to 0.16.2 with dynamic _loading_. So
  they didn't see each other in the build process and it somehow
  worked.

- With the consolidation to 0.16.6 a single cudarc is used, and
  therefore feature selection intersects with this conflict.

We can regain this back by upgrading to a different cudarc major, as
other methods require less trivial changes in docker images with regard
to CUDA library installation (having both cudarc uses with
`dynamic-linking` approach).

Signed-off-by: Dan Aloni <dan.aloni@vastdata.com>

da-x committed:
Merge branch 'main' into cudarc-fix

nv-tusharma committed:
Merge branch 'main' into cudarc-fix

nv-tusharma commented:
@da-x @rmccorm4 The issue should be resolved now, I merged the latest main changes into your branch and re-triggered the workflow. Please let me know if anything else is required, Thanks!
